src/wp-includes/capabilities.php

```diff
@@ -788,12 +788,17 @@ function author_can( $post, $capability, ...$args ) {
  * @return bool Whether the user has the given capability.
  */
 function user_can( $user, $capability, ...$args ) {
-	if ( ! is_object( $user ) ) {
-		$user = get_user_object( $user );
+	if (
+		! is_object( $user ) &&
```

Comment:
How about ` is_numeric( $user )` (or `is_int( $user )`) instead of `! is_object( $user )` here?

My reasoning is that `get_userdata()` only accepts an integer and calls `get_user_by()` internally, so any non-numeric value that isn't an object would cascade into a database query that is guaranteed to fail.